RecoTracker/LSTCore/src/alpaka/Triplet.h

```diff
@@ -561,10 +566,14 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
         // Step 2: Parallel processing of segment pairs
         for (int i = flatThreadIdxXY; i < matchCount; i += flatThreadExtent) {
           
-          int innerSegmentIndex = ranges.tripletPreselInnerOuterSgPairs()[innerLowerModuleArrayIdx][i][0];
-          int outerSegmentIndex = ranges.tripletPreselInnerOuterSgPairs()[innerLowerModuleArrayIdx][i][1];
+          //int innerSegmentIndex = ranges.tripletPreselInnerOuterSgPairs()[innerLowerModuleArrayIdx][i][0];
+          //int outerSegmentIndex = ranges.tripletPreselInnerOuterSgPairs()[innerLowerModuleArrayIdx][i][1];
           //int innerSegmentIndex = innerOuterSgPairs[i][0];
           //int outerSegmentIndex = innerOuterSgPairs[i][1];
+          unsigned int tripletIndex =
+                  ranges.tripletModuleIndices()[innerInnerLowerModuleIndex] + i;
+          unsigned int innerSegmentIndex = triplets.segmentIndices()[tripletIndex][0];
+          unsigned int outerSegmentIndex = triplets.segmentIndices()[tripletIndex][1];
```

Comment:
I'm confused about how this works, it looks like we are writing to the `triplets` SoA twice.
e.g. thread i=256 could be the first to complete and write the 0th triplet before the 0th thread reaches these lines: what will it get, it sounds like somebody else's inner/outer indices?

Reply:
Yes, you are right, I am creating a new variable.